### PR TITLE
Pulled controller code out of the rest functions

### PR DESCRIPTION
**Pull request by gaelder**

### CODE REVIEW

File: src/server/routes/controller.js [hunk 1]
```diff
@@ -100,12 +104,50 @@ function renderPNG(cyJson) {
   return renderImage(cyJson);
 }
 
+function purgeSnapshot() {
+  const tenDays = 10 * 60 * 60 * 24;
```
Comment (hmtinc) [inline src/server/routes/controller.js]: Number of days should be a parameter taken in by purgeSnapshot()

Comment (hmtinc) [inline src/server/routes/controller.js]: Number of days should be a parameter taken in by purgeSnapshot()

**Comment by d2fong** [comment]
So how does purging work?  Someone has to manually call it?

**Comment by gaelder** [comment]
It has a rest endpoint. I have a cronjob set to call it daily on my local machine, though this job needs to be created anywhere the server is run.

**Comment by d2fong** [comment]
Okay, could you also explain the reasoning for storing snapshots?  I think I missed that discussion.

**Comment by d2fong** [comment]
Okay makes sense.  Thanks.

**Comment by hmtinc** [comment]
@d2fong  Its somewhat equivalent to a temporary "edit" feature. It allows regular "view" users to make modifications and share their modifications with others without effecting that actual official layout of the graph.

**Pull request merged**